src/app/public/modules/modal-provider/types/modal-provider.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  EventEmitter
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs';
+
+import {
+  SkyModalProviderCloseArgs
+} from './modal-provider-close-args';
+
+export interface SkyModalProvider {
+
+  type: string;
+  closeCallback: Observable<SkyModalProviderCloseArgs>;
+  events?: { [key: string]: EventEmitter<any> };
+  open(paramObject?: any): any;
```

Comment:
Mind providing some loose JSDocs for the properties and interface?

Reply:
Done